### DISCUSSION TITLE
Read a database and speak results?

### DISCUSSION THREAD

**Discussion by madmacks59** [HomeSpan/HomeSpan#656] [General]
Does anyone know if the HomeKit API supports having Siri speak a text string supplied by a "device"? I what set up a Node32 to read a MariaDB table for a stored value and then have Siri respond with a spoke reply. 

Eg "Siri, what is the state of the cabin battery?"

Node32 reads the stored battery information, formats a text string, and sends that to Siri.

Siri responds "The battery is charging at 32 amps and is 80% full"...

I've looked at the HomeKit APP and I don't see any query/response type of device defined...

Thoughts?

**Comment by HomeSpan** [comment]
I'm not aware of any extensions like this, though it would be great if there was some way of enhancing Siri with custom device-specific data.

**Comment by madmacks59** [comment]
Anyone know if there's been any updates to HomeKit that might now work for the desired feature?

**Reply by HomeSpan** [reply]
How did you send something to Siri to speak?

**Reply by madmacks59** [reply]
Didn’t. But I’d like to figure out a way. If it can be figured out it would allow a lot of interesting possibilities…
